### ISSUE TITLE
NaN values when simulating Grade 3 Tincture

### ISSUE THREAD

**Issue by DarkDestry** [doxxx/ffxiv-craft-opt-web#386]
## Required Information

Class: Alchemist
Level: 80
Craftsmanship: 2480
Control: 2203
CP: 466
Recipe Name: Grade 3 Tinctures of <Anything>
Recipe Level: 80
Solver Seed: - Simulator

## Actual Behaviour

Any simulation result in NaN values for progress and quality
Solver throws the following error on console
```
TypeError: Cannot read property 'checkViolations' of undefined
    at postProgress (worker.js:295)
    at runOneGen (worker.js:272)
    at start (worker.js:259)
    at self.onmessage (worker.js:21)
```

## Steps To Reproduce

Enter the above information
Select any skill in the simulator
Note that quality and progress values are NaN

**Comment by Ranzear** [comment]
Happens with all Mastery 8 recipes.

**Comment by doxxx** [comment]
We're missing the suggested craftsmanship and control values for level 80 3-star recipes. @SaltFish-X was the one who added this feature to the code -- I'm not actually sure how those values are obtained.

**Comment by Ranzear** [comment]
It was suggested in another issue to just have a custom recipe entry method.

Is there a preferred method to provide those values though?

**Comment by doxxx** [comment]
Custom recipe entry is a long-standing request :)

If you can determine what the values should be and are comfortable with Javascript, you can create a pull request with the necessary modifications [here](https://github.com/doxxx/ffxiv-craft-opt-web/blob/master/app/js/ffxivcraftmodel.js#L2090) and [here](https://github.com/doxxx/ffxiv-craft-opt-web/blob/master/app/js/ffxivcraftmodel.js#L2543).

Otherwise, paste the values here and I can make the necessary code modifications.

**Comment by doxxx** [comment]
It looks like the Lodestone recipe pages includes this information. Maybe I can extract it from there.

**Comment by doxxx** [comment]
Although non-Master recipes only provide the recommended craftsmanship, not control, so the lookup tables are still required as fallback. :P

**Comment by Ranzear** [comment]
Craftsmanship
480: 2480

Control
480: 2195

**Comment by doxxx** [comment]
Yup that matches up with the Lodestone data. I'm working on some changes to use the Lodestone data if available, and then fallback to the lookup tables. Once the Lodestone data extraction is complete (it takes a while to download everything), I should be able to upload a fix.

**Comment by doxxx** [comment]
BTW, where did you get that information and what kind of values do you get for a non-master recipe?

**Comment by Ranzear** [comment]
![image](https://user-images.githubusercontent.com/981857/76173171-2beb1900-615a-11ea-8928-1344d68230af.png)

¯\_(ツ)_/¯

**Comment by doxxx** [comment]
I've deployed a fix for this. You will have to re-select the recipe from the dropdown after reloading the page.
